Not working on Windows

First of all, thanks for the great repo !

I tried it on Linux and it worked perfectly, but when I tried it on Windows, the configuration located in `pyproject.toml` is ignored silently...

I couldn't figure the reason why.

Is there any specific configuration to do on Windows ?

Thanks for the fast answer ! Here is my logs with `FLAKE518_DEBUG="1"` :

```
Searching for 'pyproject.toml' in C:\Users\remon\Documents\workspace\pytere
Found C:\Users\remon\Documents\workspace\pytere\pyproject.toml
Reading C:\Users\remon\Documents\workspace\pytere\pyproject.toml
Found 'tool' section in C:\Users\remon\Documents\workspace\pytere\pyproject.toml
Searching for section 'flake8' in C:\Users\remon\Documents\workspace\pytere\pyproject.toml
Found section 'flake8' in C:\Users\remon\Documents\workspace\pytere\pyproject.toml
Searching for section 'flake518' in C:\Users\remon\Documents\workspace\pytere\pyproject.toml
Section 'flake518' does not exist in C:\Users\remon\Documents\workspace\pytere\pyproject.toml
Found entries in C:\Users\remon\Documents\workspace\pytere\pyproject.toml. Adding additional configuration...
The following configuration is written: {'flake8': {'max-line-length': 119, 'max-complexity': 10, 'per-file-ignores': '__init__.py:F401', 'extend-ignore': 'E203'}} 
Using additional configuration file 'C:\Users\remon\Documents\workspace\pytere\.flake518_itzm8qeh.cfg' for flake8 call. File will be deleted afterwards
The following arguments are applied now: ['--config', 'C:\\Users\\remon\\Documents\\workspace\\pytere\\.flake518_itzm8qeh.cfg']
.\setup.py:8:80: E501 line too long (109 > 79 characters)
.\setup.py:9:80: E501 line too long (119 > 79 characters)
.\setup.py:10:80: E501 line too long (98 > 79 characters)
.\setup.py:17:80: E501 line too long (116 > 79 characters)
.\setup.py:18:80: E501 line too long (119 > 79 characters)
.\setup.py:19:80: E501 line too long (110 > 79 characters)
.\setup.py:22:80: E501 line too long (101 > 79 characters)
.\setup.py:35:80: E501 line too long (101 > 79 characters)
```

---

As you can see it's weird, the configuration is correctly detected (`'max-line-length': 119`), but flake8 complains about too long lines...

--- a/.changelogs/v1.2.3.md
+++ b/.changelogs/v1.2.3.md
@@ -1,0 +1,5 @@
+# Changelog
+
+## Bug-fixes
+
+* On Windows file systems NamedTemporaryFiles should not be deleted. Hence the file is not read by flake8 properly.

--- a/.changelogs/v1.2.3.md
+++ b/.changelogs/v1.2.3.md
@@ -2,4 +2,4 @@
 
 ## Bug-fixes
 
-* On Windows file systems NamedTemporaryFiles should not be deleted. Hence the file is not read by flake8 properly.
+* On Windows file systems NamedTemporaryFiles should not be deleted. Hence the file is not read by flake8 properly. (see [#8](https://github.com/carstencodes/flake518/issues/8))

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -6,17 +6,6 @@ on:
       - 'v*'
 
 jobs:
-  changelog:
-    name: Generate changelog
-    runs-on: ubuntu-latest
-    outputs:
-      changelog: ${{ steps.changelog.outputs.changelog }}
-    steps:
-      - name: Generate changelog
-        id: changelog
-        uses: heinrichreimer/github-changelog-generator-action@v2.1.1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
   deploy:
     name: Deploy wheel to pypi
     runs-on: ubuntu-latest
@@ -43,13 +32,28 @@ jobs:
   release:
     name: Create Release
     runs-on: ubuntu-latest
-    needs: [ changelog, deploy ]
+    needs: [ deploy ]
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
       - name: Get tag name
         uses: battila7/get-version-action@v2
         id: tag_name
+      - name: Generate changelog
+        id: changelog
+        run: |
+          chl_file=".changelogs/${{steps.tag_name.outputs.version}}.md"
+          if [ -f "${chl_file}" ]
+          then
+            message=$(cat "${chl_file}")
+            message="${message//'%'/'%25'}"
+            message="${message//$'\n'/'%0A'}"
+            message="${message//$'\r'/'%0D'}"
+          else
+            changelog="No changelog provided"
+          fi
+
+          echo ::set-output name=changelog::${changelog}
       - name: Create Release
         id: create_release
         uses: actions/create-release@v1
@@ -60,6 +64,6 @@ jobs:
         with:
           tag_name: ${{ steps.tag_name.outputs.version }}
           release_name: Release ${{ steps.tag_name.outputs.version }}
-          body: ${{ needs.changelog.outputs.changelog }}
+          body: ${{ steps.changelog.outputs.changelog }}
           draft: false
           prerelease: ${{ (startsWith(steps.tag_name.outputs.version, 'v0')) || (endsWith(steps.tag_name.outputs.version, '-pre')) || (endsWith(steps.tag_name.outputs.version, '-dev')) }}

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [project]
 name = "flake518"
-version = "1.2.2"
+version = "1.2.3"
 description = "A small wrapper around flake8 to support PEP518 pyproject.toml as configuration file."
 authors = [
     {name = "Carsten Igel", email = "cig@bite-that-bit.de"},

--- a/src/flake518/adapter.py
+++ b/src/flake518/adapter.py
@@ -237,6 +237,8 @@ def run(argv: Optional[_List[str]] = None) -> None:
         args.append("-vv")
     elif _get_os_var_value("FLAKE518_VERBOSE"):
         args.append("-v")
+    
+    config_file: Optional[Path] = None
         
     if (len(config)) > 0:
         logger.debug(
@@ -244,9 +246,8 @@ def run(argv: Optional[_List[str]] = None) -> None:
             py_project,
         )
         update_configuration(config)
-        delete_file = not keep_file
         with NamedTemporaryFile(
-            "x+t", prefix=".flake518_", suffix=".cfg", delete=delete_file, dir=py_project.parent
+            "x+t", prefix=".flake518_", suffix=".cfg", delete=False, dir=py_project.parent
         ) as handle:
             write_config_to_ini(config, handle)
             logger.debug(
@@ -258,8 +259,19 @@ def run(argv: Optional[_List[str]] = None) -> None:
             args.append("--config")
             args.append("{}".format(handle.name))
             logger.debug("The following arguments are applied now: %s", args)
-            run_flake8(args)
-
+            config_file = Path(handle.name)
     else:
         logger.debug("Running flake8 without modified configuration")
+    
+    try:
         run_flake8(args)
+    except SystemExit:
+        if not keep_file and \
+                (config_file is not None and config_file.exists() and config_file.is_file()):
+            config_file.unlink()
+        raise
+    else:
+        if not keep_file and \
+                (config_file is not None and config_file.exists() and config_file.is_file()):
+            config_file.unlink()
+            

